File: 3025-Find-the-Number-of-Ways-to-Place-People-I/solution.java
import java.util.*;

class Solution {
    public int numberOfPairs(int[][] points) {
        Arrays.sort(points, (a, b) -> a[0] == b[0] ? b[1] - a[1] : a[0] - b[0]);
        int n = points.length;
        int ans = 0;
        for (int i = 0; i < n; i++) {
            int yi = points[i][1];
            int maxY = Integer.MIN_VALUE;
            for (int j = i + 1; j < n; j++) {
                int yj = points[j][1];
                if (yj <= yi && yj > maxY) {
                    ans++;
                    maxY = yj;
                }
            }
        }
        return ans;
    }
};

File: 3025-Find-the-Number-of-Ways-to-Place-People-I/solution.py
from typing import List

class Solution:
    def numberOfPairs(self, points: List[List[int]]) -> int:
        points.sort(key=lambda p: (p[0], -p[1]))
        n = len(points)
        ans = 0
        for i in range(n):
            yi = points[i][1]
            maxY = float('-inf')
            for j in range(i + 1, n):
                yj = points[j][1]
                if yj <= yi and yj > maxY:
                    ans += 1
                    maxY = yj
        return ans